src/wp-admin/includes/image.php

```diff
@@ -301,7 +301,11 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 			// The image may need to be converted regardless of its dimensions.
 			$output_format = wp_get_image_editor_output_format( $file, $imagesize['mime'] );
 
-			if ( is_array( $output_format ) && array_key_exists( $imagesize['mime'], $output_format ) ) {
+			if (
+				is_array( $output_format ) &&
+				array_key_exists( $imagesize['mime'], $output_format ) &&
+				$output_format[ $imagesize['mime'] ] !== $imagesize['mime']
```

Comment:
Very nice, as a maybe I am wondering if WP should check this after the filter runs in `wp_get_image_editor_output_format()` but I don't think it's a blocker.

Reply:
Potentially, though that goes back to the question what the `image_editor_output_format` filter is supposed to be for.

"output format" != "conversion format", so the filter name doesn't imply that the output format needs to be different than the input format.

Reply:
> "output format" != "conversion format", so the filter name doesn't imply that the output format needs to be different than the input format.

Works for me, thanks Felix. I'll commit this shortly.

Reply:
I just beat you to it :)
https://core.trac.wordpress.org/changeset/59366

Reply:
> wondering if WP should check this after the filter runs in wp_get_image_editor_output_format()

Was thinking the same, but agree that the function name and filter name don't suggest that this is only for converting images.

Perhaps this can be mentioned in the docblock. Maybe something like:
> Note that the output format for a particular mime-type may be the same as the format of the actual image.